src/configuration.js

```diff
@@ -0,0 +1,161 @@
+            const dir = join(process.cwd(), './apify_storage');
+            this.set('localStorageDir', dir);
+            log.warning(`Neither ${ENV_VARS.LOCAL_STORAGE_DIR} nor ${ENV_VARS.TOKEN} environment variable is set, defaulting to ${ENV_VARS.LOCAL_STORAGE_DIR}="${dir}"`);
+        }
+    }
+
+    /**
+     * @param {string} key
+     * @param {string | number | boolean} [defaultValue]
+     * @return {string | number | boolean}
+     */
+    get(key, defaultValue) {
+        // prefer env vars
+        const envKey = Configuration.ENV_MAP_REVERSED[key] ?? key;
+        const envValue = process.env[envKey] ?? process.env[ENV_VARS[envKey]];
+
+        if (envValue != null) {
+            if (envKey in INTEGER_ENV_VARS) {
+                return +envValue;
+            }
+
+            if (envKey in Configuration.BOOLEAN_ENV_VARS) {
+                // 0, false and empty string are considered falsy values
+                return !['0', 'false', ''].includes(envValue.toLowerCase());
+            }
+
+            return envValue;
+        }
+
+        // check instance level options
+        if (key in this.options) {
+            return this.options[key];
+        }
+
+        // fallback to defaults
+        return defaultValue ?? Configuration.DEFAULTS[key];
+    }
+
+    /**
+     * @param {string} key
+     * @param {string | number} value
+     */
+    set(key, value) {
+        this.options[key] = value;
+    }
+
+    /**
+     * Creates an instance of ApifyClient using options as defined in the environment variables.
+     * @param {object} [options]
+     * @param {string} [options.token]
+     * @param {string} [options.maxRetries]
+     * @param {string} [options.minDelayBetweenRetriesMillis]
+     * @return {ApifyClient}
+     * @internal
+     */
+    createClient(options = {}) {
+        return new ApifyClient({
+            baseUrl: this.get('apiBaseUrl'),
+            token: this.get('token'),
+            ...options, // allow overriding the instance configuration
+        });
+    }
+
+    /**
+     * Creates an instance of ApifyStorageLocal using options as defined in the environment variables.
+     * @param {object} [options]
+     * @param {string} [options.storageDir]
+     * @param {boolean} [options.enableWalMode=true]
+     * @return {ApifyStorageLocal}
+     * @internal
+     */
+    createLocalStorage(options = {}) {
+        const storageDir = options.storageDir ?? this.get('localStorageDir');
+        const enableWalMode = options.enableWalMode ?? this.get('localStorageEnableWalMode');
+        const storage = new ApifyStorageLocal({ ...options, storageDir, enableWalMode });
+
+        process.on('exit', () => {
+            // TODO this is not public API, need to update
+            // storage local with some teardown
+            storage.dbConnections.closeAllConnections();
+        });
+
+        return storage;
+    }
+
+    /**
+     * @return {Configuration}
+     */
+    static getDefaults() {
```

Comment:
I'm not a big fan of the name, because as I looked through the code, it's mostly used as a `getConfig()` aka, give me the current configuration. Not give me the configuration defaults.

Reply:
So I went for `getGlobalConfig()` to be explicit about what it really returns. Also added jsdoc that points to its counterpart `sdk.config`, will be adding more docs about probably to the `Apify` class, so will link it once its there.